cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction/Pretty.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.CLI.Shelley.Run.Transaction.Pretty (prettyTx) where
+
+import           Cardano.Api
+                   (ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley))
+import           Cardano.Api.Byron (TxBody (ByronTxBody))
+import           Cardano.Api.Shelley (TxBody (ShelleyTxBody))
+import           Cardano.Prelude
+import           Data.Aeson (object, (.=))
+import           Data.Aeson.Encode.Pretty (Config (confCompare), defConfig, encodePretty')
+import qualified Shelley.Spec.Ledger.API as Shelley
+import qualified Cardano.Ledger.ShelleyMA.TxBody as ShelleyMA
+
+prettyTx :: TxBody era -> LByteString
+prettyTx body0 =
+  encodePretty' defConfig{confCompare = compare} $
+  object $
+  case body0 of
+    ByronTxBody tx -> ["era" .= ("Byron" :: Text), "tx" .= tx]
+    ShelleyTxBody ShelleyBasedEraShelley body aux ->
+      [ "era" .= ("Shelley" :: Text)
+      , "inputs" .= _inputs
+      , "outputs" .= _outputs
+      , "certificates" .= fmap showText _certs
+      , "withdrawals" .= withdrawals
+      , "fee" .= _txfee
+      , "timetolive" .= _ttl
+      , "update" .= fmap showText _txUpdate
+      , "metadata_hash" .= fmap showText _mdHash
+      , "auxiliary_data" .= fmap showText aux
+      ]
+      where
+        Shelley.TxBody{_inputs, _outputs, _certs, _wdrls, _txfee, _ttl, _txUpdate, _mdHash} =
+          body
+        Shelley.Wdrl withdrawals = _wdrls
+    ShelleyTxBody ShelleyBasedEraAllegra body aux ->
+      [ "era" .= ("Allegra" :: Text)
+      , "inputs" .= inputs
+      , "outputs" .= outputs
+      , "certificates" .= fmap showText certificates
+      , "withdrawals" .= withdrawals
+      , "fee" .= txfee
+      , "validity_interval" .= showText validity
+      , "update" .= fmap showText update
+      , "auxiliary_data_hash" .= fmap showText adHash
+      , "mint" .= mint
+      , "auxiliary_data" .= fmap showText aux
+      ]
+      where
+        ShelleyMA.TxBody inputs outputs certificates wdrls txfee validity update adHash mint =
+          body
+        Shelley.Wdrl withdrawals = wdrls
+    ShelleyTxBody ShelleyBasedEraMary _body _aux -> ["era" .= ("Mary" :: Text)]
+
+showText :: Show a => a -> Text
```

Comment:
We already have `textShow` in `Cardano.CLI.Helpers`

Reply:
thanks